src/Shared/EscapingUtilities.cs

```diff
@@ -72,21 +52,37 @@ internal static string UnescapeAll
             int indexOfPercent = escapedString.IndexOf('%');
             if (indexOfPercent == -1)
             {
-                return escapedString;
+                return trim ? escapedString.Trim() : escapedString;
             }
 
             // This is where we're going to build up the final string to return to the caller.
             StringBuilder unescapedString = StringBuilderCache.Acquire(escapedString.Length);
 
             int currentPosition = 0;
+            int escapedStringLength = escapedString.Length;
+            if (trim)
+            {
+                while (currentPosition < escapedString.Length && Char.IsWhiteSpace(escapedString[currentPosition]))
```

Comment:
Wouldn't it be better to just wait until we can use Spans here, instead of doing it by hand?

Reply:
Since not everyone wants to trim it, that would make it more complicated rather than cleaner in my opinion.